.github/workflows/run-e2e-tests.yml

```diff
@@ -33,13 +33,19 @@ jobs:
   e2e:
     runs-on: ubuntu-latest
     strategy:
-      max-parallel: 3
+      max-parallel: 4
       matrix:
-        java: [ 8, 11, 17 ]
+        java: [ 8, 11, 17, 21 ]
     name: End-to-end tests java${{ matrix.java }}
     env:
-      JAVA_VERSION: ${{ matrix.java }}
       AWS_DEFAULT_REGION: eu-west-1
+
+      # If matrix.version is 21, use 17, otherwise use matrix.version
+      # This is because AspectJ does not yet support weaving with Java21; we want
+      # to test the Java21 runtime, but we can't yet use the JDK21 compiler.
+      # https://github.com/eclipse-aspectj/aspectj/issues/260#issuecomment-1815920274
+      JAVA_VERSION: ${{ (matrix.java == 21 && '17') || matrix.java }}
```

Comment:
we should remove that...

Reply:
We should remove it in main, this is just a branch merge